tmail-backend/jmap/extensions-es-v7/src/main/java/com/linagora/tmail/james/jmap/ESEmailAddressContactSearchEngine.java

```diff
@@ -120,6 +124,76 @@ public Publisher<EmailAddressContact> autoComplete(AccountId accountId, String p
             .map(Throwing.function(this::extractContentFromHit).sneakyThrow());
     }
 
+    @Override
+    public Publisher<EmailAddressContact> list(AccountId accountId) {
+        SearchRequest request = new SearchRequest(configuration.getUserContactReadAliasName().getValue())
+            .source(new SearchSourceBuilder()
+                .query(QueryBuilders.boolQuery()
+                    .should(QueryBuilders.termQuery(ACCOUNT_ID, accountId.getIdentifier()))
+                    .minimumShouldMatch(1)));
+
+        return client.search(request, RequestOptions.DEFAULT)
+            .flatMapIterable(searchResponse -> ImmutableList.copyOf(searchResponse.getHits().getHits()))
+            .map(Throwing.function(this::extractContentFromHit).sneakyThrow());
+    }
+
+    @Override
+    public Publisher<EmailAddressContact> list(Domain domain) {
+        SearchRequest request = new SearchRequest(configuration.getDomainContactReadAliasName().getValue())
+            .source(new SearchSourceBuilder()
+                .query(QueryBuilders.boolQuery()
+                    .should(QueryBuilders.termQuery(DOMAIN, domain.asString()))
+                    .minimumShouldMatch(1)));
+
+        return client.search(request, RequestOptions.DEFAULT)
+            .flatMapIterable(searchResponse -> ImmutableList.copyOf(searchResponse.getHits().getHits()))
+            .map(Throwing.function(this::extractContentFromHit).sneakyThrow());
+    }
```

Comment:
To do a full listingm a simple search request is not enough as the count of elements is bounded.

You need to have a scrolled search here? Maybe we could have a test trying to list 1000+ contacts...